## Dictionary Maker

Starting off with [Ogden's Basic Wordlist](http://ogden.basic-english.org/words.html), adding in contractions for later (and after the fact)

In [111]:
def get_basic_english():
        '''
        Creates a list of basic english words from a csv file containing
        Ogden's 850 basic english words.
        '''
        # Convert basic english words to a list
        basic_english_df = pd.read_csv('../data/basic_english_wordlist.csv')
        print basic_english_df.head()
        basic_english = []
        basic_english = [a for a in basic_english_df['WORD']]
        # add the various conjugations of 'to be' and 'a'
        basic_english.append('an')
        basic_english.append('is')
        basic_english.append('was')
        basic_english.append('are')
        basic_english.append('were')
        basic_english.append('they')
        # 'I' causes weird issues...
        basic_english[basic_english.index('I')] = 'big'
        basic_english.append('she')
        basic_english.append('hers')
        basic_english.append('his')
        basic_english.append('my')
        basic_english.append('him')
        basic_english.append('her')
        basic_english.append('your')
        basic_english.append('their')
        basic_english.append('might')
        basic_english.append('must')
        basic_english.append('can')
        basic_english.append('did')
        basic_english.append('could')
        basic_english.append('should')
        basic_english.append('would')
        basic_english.append('that')
        basic_english.append('what')
        basic_english.append('we')
        basic_english.append('small')
        basic_english[basic_english.index('colour')] = 'color'
        print 'basic_english has {} words'.format(len(basic_english))
        return basic_english

In [112]:
import pandas as pd
basic_english = get_basic_english()

      WORD
0        a
1     able
2    about
3  account
4     acid
basic_english has 875 words


So, 875 is not a lot of words, and even CK Ogden noted that each industry will need around 150 extra words to communicate in that industry.  So, first things first, let's grab the [internaion words](http://ogden.basic-english.org/intlword.html) that the Ogden crowd already likes...

In [113]:
new_words = 'international, '
# International Nouns
new_words = new_words + 'alcohol, aluminum, automobile, bank, bar, beef, beer, calendar, check, chemist, chocolate, chorus, cigarette, club, coffee, colony, dance, engineer, gas, hotel, influenza, lava, madam, nickel, opera, orchestra, paraffin, park, passport, patent, phonograph, piano, police, post, program, propaganda, radio, restaurant, sir, sport, taxi, tea, telegram, telephone, terrace, theater, tobacco, university, whisky, zinc, '
# Names of sciences
new_words = new_words + 'Algebra, Arithmetic, Biology, Chemistry, Geography, Geology, Geometry, Mathematics, Physics, Physiology, Psychology, Zoology, '
#International names used in titles, organizations, diplomacy, etc...
new_words = new_words + 'College, Dominion, Embassy, Empire, Imperial, King, Museum, President, Prince, Princess, Queen, Royal, '
# general utility...
new_words = new_words + 'ammonia, asbestos, autobus, ballet, cafe, catarrh, champagne, chauffeur, circus, citron, cocktail, cognac, dynamite, encyclopedia, glycerin, hyena, hygiene, hysteria, inferno, jazz, liqueur, macaroni, malaria, mania, nicotine, olive, omelet, opium, paradise, penguin, platinum, potash, pajamas, pyramid, quinine, radium, referendum, rheumatism, rum, salad, sardine, tapioca, toast, torpedo, vanilla, violin, visa, vodka, volt, zebra, '
# Onomatopoeic (new word just added to my vocab right there)
new_words = new_words + 'bang, hiss, meow, purr, quack, bark, woof'  # last two aded by me, dogs ftw
# time and numbers
new_words = new_words + 'January, February, March, April, May, June, July, August, September, October, November, December, '
new_words = new_words + 'Monday, Tuesday, Wednesday, Thursday, Friday, Saturday, Sunday, '
new_words = new_words + 'One, two, three, four, five, six, seven, eight, nine, ten, eleven, twelve, thirteen, fourteen, fifteen, sixteen, twenty, twenty-one, thirty, forty, fifty, hundred, thousand, million, once, twice, half, quarter, third, fourth, fifth, '
# New century skipped for now, add later

#the science list
new_words = new_words + 'active, adjacent, adsorption, age, alternative, application, arc, area, arrangement, ash, axis, break, bubble, capacity, case, cell, column, component, compound, continuous, cross, decrease, deficiency, deposit, determining, difference, difficulty, direct, disappearance, discharge, disturbance, elimination, environment, equation, evaporation, exact, experiment, explanation, focus, friction, fusion, generation, groove, guard, hinge, impurity, individual, interpretation, investigation, joint, latitude, layer, length, link, longitude, mean, melt, mixture, nucleus, origin, path, per, pressure, projection, proof, reference, relative, reproduction, resistance, rigidity, rock, rot, rotation, screen, seal, section, sensitivity, shadow, shear, shell, similarity, solution, spark, specialization, specimen, stimulus, strain, strength, stress, substitution, successive, supply, surface, swelling, thickness, thrust, tide, transmission, transparent, tube, valve, '
new_words = new_words + 'anesthetic, bomb, electricity, magnetic, microscope, neutron, organism, petroleum, serum, thermometer, tropism, turbine, vitamin, '
new_words = new_words + 'allotropy, endothelium, hemocyanin, isomorphism, phylogeny, tropism, '
# phys sci
new_words = new_words + 'absorption, beaker, buoyancy, charge, circuit, clip, coil, collision, combination, conductor, conservation, corrosion, density, dilution, dissipation, explosion, flask, fume, funnel, furnace, grating, image, insulator, lag, lens, medium, oxidation, particle, pendulum, plug, porcelain, radiation, reagent, receiver, reflux, repulsion, residue, reversible, saturated, solvent, stable, stream, suspension, switch, tongs, trap, valency, vapor, vortex, wedge, '
# geology
new_words = new_words + 'accessory, birefringence, cast, cave, clay, cleavage, contour, desert, dike, dip, drift, erosion, eruption, extinction, fan, fault, flint, flood, flow, foliation, fracture. glacier, gravel, ground, hill, igneous, inclusion, intercept, interpenetration, intrusion, lake, limestone, mud, ore, outcrop, outlier, overlap, plain, scarp, schist, sedimentary, shale, shore, sill, slate, strike, texture, twin, unconformity, valley, '
# math & mechanics
new_words = new_words + 'acceleration, amplitude, approximation, arbitrary, circumference, congruent, constant, cusp, damping, denominator, diameter, divisor, fraction, infinity, integer, intersection, inverse, lever, load, locus, magnitude, momentum, multiple, multiplication, node, numerator, oblique, piston, prime, probability, product, projectile, pulley, quantity, quotient, ratio, reciprocal, rectangle, recurring, reinforcement, resolution, subtraction, sum, term, total, triangle, unknown, variable, velocity, volume, '
# biology
new_words = new_words + 'abdomen, appendage, bark, beak, bud, cartilage, cavity, claw, climber, creeper, degenerate, domesticating, duct, ferment, fertilizing, fiber, fin, fresh, germinating, gill, gland, hoof, host, inheritance, jaw, juice, kidney, liver, lung, mature, metabolism, parent, petal, pollen, sac, scale, secretion, sepal, skull, slide, soil, stain, stalk, stamen, sucker, thorax, tissue, vascular, vestigial, wild, '
# commerece
new_words = new_words + 'acceptance, address, agency, allowance, assets, assistant, average, bale, bankrupt, barrel, bill, broker, certificate, charge, claim, client, code, complaint, consignment, cost, court, customs, debit, deck, defect, delivery, export, gross, guarantee, hire, hold, import, investment, liability, license, load, loan, packing, pair, partner, purchase, reference, retail, sale, sample, show, sight, strike, supply, wholesale, '
# econ
new_words = new_words + 'accident, arbitration, asset, average, bill, broker, budget, circulation, combine, consumer, conversion, correlation, cost, deflation, demand, deposit, discount, efficiency, effort, employer, experiment, factor, fatigue, guarantee, habit, index, inflation, investment, liability, loan, margin, monopoly, partner, pension, plan, population, purchase, rent, retail, sale, saving, security, service, share, speculation, statistics, stimulus, strike, supply, wholesale, '
# buis and econ
new_words = new_words + 'asset, average, bill, broker, cost, guarantee, investment, liability, loan, partner, purchase, retail, sale, strike, supply, wholesale, '
new_words = new_words + 'capital, firm, wages, '
# Trade, despite the greyed-outness of the text
new_words = new_words + 'cash, premium, service, slump, dock, net, warrant, '
new_words = new_words + 'canal, capital, carton, consul, guarantee, minus, percent, plus, propaganda, stand, wagon, '
new_words = new_words + 'balance, forward, case, royalty, duties, '

# Verse
#coincidently handles some words I was worried about for Dumas, like sword
# God/god1 thing -> god, since capitalization would be maintained thru other bits of code.
new_words = new_words + 'angel, arrow, beast, blind, bow, breast, bride, brow, bud, calm, child, cross, crown, curse, dawn, delight, dew, dove, dream, eagle, eternal, evening, evil, fair, faith, fate, feast, flock, flow, fountain, fox, gentle, glad, glory, god, grace, grape, grief, guest, hawk, heaven, hell, hill, holy, honey, honor, image, ivory, joy, lamb, lark, life, lion, lord, meadow, melody, mercy, noble, passion, perfume, pity, pool, praise, prayer, pride, priest, purple, rapture, raven, robe, rock, rose, rush, search, shining, shower, sorrow, soul, spear, spirit, storm, stream, strength, sword, thief, tower, travel, valley, veil, vine, violet, virgin, virtue, vision, wandering, wealth, weariness, weeping, wisdom, wolf, wonder, '

# Bible
new_words = new_words + 'altar, ark, ass, ax, baptism, blessing, captain, cattle, circumcision, cock, deceit, disciple, envy, flesh, forgiveness, generation, herd, heritage, husband, kingdom, leaven, leper, locust, master, neighbor, oath, ox, people, pillar, preaching, prophet, revelation, righteousness, saint, salvation, savior, scribe, sin, spice, tent, testament, thorn, tribe, virgin, widow, wife, witness, world, worship, wrath, yoke, '

# Social Science... funny, since making up new words is their thing
new_words = new_words = 'heredity, incentive, integration, standardization, State, technology, unemployment, environment, association, attitudes, class, community, culture, custom, '
new_words = new_words = 'capital, firm, wages '


new_words = new_words = 'decentralization, structure, concentration, control, will, '


# Net words...
# note, many repeats, I will be 'set'ing these later, so i include all to make sure I don't miss one
new_words = new_words + 'alphabet, back , bar , base , board , body , boot , box , broken , bulletin , button , cable , card , check , clear , center , connection , control , copy , cord , current , cut , down , dictionary , division , drop , end , engine , field , fixed , folder , forward , group , hard , head , help, in , ink , join , key , knowledge , language , last , left , library , line , list , lock , log, machine , map , memo , memory , move , name , net , news , observation , off , office , on , open , out , page , parallel, park , person , pin , pointer , post , print , program , process , public , pull , push , put , range , reading , record , right , roll , run , send , shut , size , smooth, soft , sort , sound , space , start , statement , step , sticky , store , support , syntax , system , table , test , text , thread , tray , unit , up , value , view , voice , window , wire , word , work, '
new_words = new_words + 'backspace , bookmark, input , keyboard , lockout , lockup, lookup, online , output , network , readme , shutdown , startup , update, '
# these were two-word expressions, want to make sure I get both words seperatly and together
new_words = new_words + 'back, button, black, box, cold, start, high, memory, news, group, number, lock, page, up, page, down, power, cord, space, bar, support, line, thumb, drive, '

# including candidates
new_words = new_words + ' analog , archive , base , center , client , column , computer , control , copy , cyberspace , disk , diskette , division , hotlink , icon , information , internet , list , logic , memory , menu , modem , page , parallel , personal , personal computer , PC , post , program , project , protocol , server , system , table , test , text , type , variable , virtual , website, '
new_words = new_words + 'code , connection , data , digital , domain , homepage , insert , monitor , packet , park , value , webmaster, '
new_words = new_words + 'base, center, copy, list, page, post, program, project, system, table, test, '


# even though NOT basic words, adding them anyway
new_words = new_words + 'address, align, analog, center, click, code, column, computer, cyber, data, delete, digital, director, disk, domain, drag, file, font, hold, home, hyper, information, insert, internet, link, load, logic, mail, menu, modem, monitor, mouse, packet, protocol, recent, reply, row, search, serial, site, text, type, web, '

stuff = 'absence , accident [b] , address [b] , adventure , advice , age[s] , agent , ago , along , also , always , ambition , anchor , ankle , arrangements [s], ash [s] , awkward , balcony , barrel [b] , beard , beat , behind , belt , bet , blame , blanket , both , bottom , brave , breakfast , breast [v] , bubble [s] , bud [v] , bunch , burial , busy , calculation , call , capital [g] , carpet , case [s] , ceremony , chair , character , child [v] , chimney , china , choice , civilization , clay , clever , collection , column [s,c] , communications , concept , concrete , convenient , cool , corner , cost [b] , crop , cross[s,v] , cunning , date , demand [b] , department , dew [v] , difficulty [s] , dinner , disgrace , dislike , ditch , dive , divorce , doll , dreadful , dream [v] , dull , duty , each , easy , economy , effort [b] , either , employer [b] , empty , enemy , envelope , envy , evening [v] , exact [s] , excitement , exercise , explanation [s] , explosion , expression , eyebrow , eyelash , failure , fair [v] , famous , fan , fastening , fever , figure , financial , flash , flesh , flood , flour , forecast , forehead , foreign , forgiveness , fresh , frost , frozen , funny , fur , furniture , gate , generation [s] , germ , god [v] , grand , grateful , grease , grief [v] , grocery , ground , guard [s] , guess , gum , habit [b] , handkerchief , handle , heavy , hedge , hil l[v] , holiday , home , honest , honey [v] , human , hunt , hurry , hurt , husband , imagination , innocent , institution , intelligent , invitation , jam , jaw , jealous , jerk , joint [s] , jug , juice , jury , justice , kennel , kidney , kitchen , knock , lace , lame , lamp , large , lawyer , lazy , lecture , legal , lesson , lever , license , lid , life [v] , lime , liver , load [b] , local , loop , luck , lump , lunch , lung , magic , manner , many , marble , marriage , mast , mattress , mean [s] , meaning , medicine , member , mess , message , mill , mineral , model , modern , modest , mood , moral , moustache , mud , murder , nasty , nature , navy , neat , neglect , neighbor , nest , next , nice , nostril , nurse , obedient , officer , orchestra , organ , origin [s] , oval , own , pad , pair [b] , pan , paragraph , party , passage , patience , pedal , people , perfect , plaster , poetry , pool [v] , practice , praise [v] , prayer [v] , prick , priest [v] , progress , promise , proof [s] , proud [v] , pupil , pure , race , real , reference [s,t] , remark , remedy , repair , result , revenge , rich , rise , rival , rude , rust , sale [b] , satisfaction , saucer , scratch , screen [s] , search , selfish , sentence , set , shave , sheet , shell [s] , shoulder , show [b] , sight [b], since , sleeve , social , soldier , sorry , spark [s] , spirit [v] , spit , splash , spot , stable, stair, steady , storm [v] , strain , straw , stream [v] , string , study , subject , success , sum , supply [s,b] , surface [s] , surgeon , suspicious , swing , sympathetic , tailor , tame , tap , tear , tent , thief [v] , thimble , threat , tie , too , towel , tower [v] , traffic , tragedy , trap , travel [v] , treatment , truck , tube [s], tune , tunnel , twin , typist , ugly , understanding , universe , valle y[v] , vegetable , victim , victory , vote , weak , welcome , whether , widow , wife , wild , world , wreck , wrist , yawn'
new_words = new_words + stuff.replace(' [b]','').replace('[s]','').replace('[s,c]','').replace('[g]','').replace('[v]','').replace('[s,b]','').replace('[s,t]','').replace(' ,',',')+', '

# doubles...
new_words = new_words + 'accident , age , agent , arrangement , ash , barrel , breast , bubble , bud , capital , case , child , client , column , cost , cross , demand , deposit , dew , difficulty , dream , effort , employer , evening , exact , experiment , explanation, fair , god , grief , guard , habit , hill , honey , joint , life , load , mean , origin , pool, praise , prayer, priest , proof , proud , reference , rock , sale , screen , search , shell , show , sight , spark , spirit , storm , stream , strength , supply , surface , thief , tower , travel , tube , valley , '
new_words = new_words + 'can' # sure, I'll bite
new_words.replace('trousers','pants') # 'Merica!

expanded_english = new_words.split(', ')
basic_english = basic_english + list(set(expanded_english))
basic_english.append('club')
basic_english.append('pistol')
print basic_english[:10]
basic_english.sort()
basic_english = list(set([a.replace(' ','').lower() for a in basic_english]))
print "Now we have {} words".format(len(basic_english))



['a', 'able', 'about', 'account', 'acid', 'across', 'act', 'addition', 'adjustment', 'advertisement']
Now we have 1315 words


Include the [most common](https://en.wikipedia.org/wiki/Most_common_words_in_English) words...


In [114]:
add = "time person year way day thing man world life hand part child eye woman place work week case point government company \
number group problem fact be have do say get make go know take see come think look want give use find tell ask work seem feel \
try leave call good new first last long great little own other old right big high different small large next early young \
important few public bad same able to of in for on with at by from up about into over after beneath under above the and \
a that I it not he as you this but his they her she or an will my one all would there their"
add_lst = add.split(' ')
basic_english =  basic_english + add_lst
basic_english.sort()
basic_english = list(set([a.replace(' ','').lower() for a in basic_english]))
print "Now we have {} words".format(len(basic_english))

Now we have 1332 words


In [108]:
import gensim
model = gensim.models.KeyedVectors.load_word2vec_format('../model/GoogleNews-vectors-negative300.bin', binary=True)

In [115]:
import nltk.data
from nltk import pos_tag, word_tokenize
from nltk.stem.lancaster import LancasterStemmer
def make_new_dictionary():
        '''
        This means the data file associated with this particular instance does
        not have a 'data/basic_english.pickle' file associated with it, and
        we need to make one.
        '''
        print 'Running...'
        vocab_google = model.vocab.keys()

        print len(basic_english)
        # adding contractions...
        contractions_df = pd.read_csv('../data/contractions.csv', sep=' -')
        contractions = [word for word in contractions_df['from']]
        contractions[18] = "mightn't"
        st = LancasterStemmer()
        stem_gn = [st.stem(key) for key in model.vocab.keys()]
        stem_se = [st.stem(word) for word in basic_english]
        my_dict = {}
        stems = []
        threshold = 0.2  # much smaller and it grabs weird things
        print 'starting...'
        for sim_in in xrange(len(basic_english)):
            indices = []
            check = []
            if (len(basic_english[sim_in]) >= 2) and (stem_se[sim_in] not in stems):
                stems.append(stem_se[sim_in])
                indices = [i for i, s in enumerate(stem_gn)
                           if stem_se[sim_in] == s]
                check = [i for i, s in enumerate(vocab_google)
                         if basic_english[sim_in] == s]
                print '{}: {} of {}'.format(basic_english[sim_in], sim_in, len(basic_english))
                # check, indices
                if len(check) > 0:
                    for index in indices:
                        a = threshold
                        if (model.similarity(basic_english[sim_in], vocab_google[index]) >= a):
                            my_dict[vocab_google[index].lower()] = \
                                    [vocab_google[index].lower(),
                                     basic_english[sim_in].lower(),
                                     pos_tag([vocab_google[index].lower()])[0][1]]
                            print "    {} -> {}".format(vocab_google[index].lower(), my_dict[vocab_google[index].lower()])
                        # add itself last... to overwrite issues w/above
                    a = basic_english[sim_in].lower()
                    b = pos_tag([a.lower()])[0][1]
                    my_dict[basic_english[sim_in].lower()] = [a, a, b]
        my_dict['i'] = ['i', 'i', pos_tag(['i'])[0][1]]  # add 'I
        my_dict['a'] = ['a', 'a', pos_tag(['a'])[0][1]]  # add 'I
        basic_english.append('i')
        for word in basic_english:
            wordy = word
            if len(word) <= 1:
                wordy = word + "'"
            for con in contractions:
                if wordy.lower() in con.lower()[0:len(wordy)]:
                    my_dict[con.lower()] = [con.lower(), wordy.lower(),
                                            pos_tag([wordy.lower()])[0][1]]
        # Save it...
        with open('../data/basic_english.pickle', 'wb') as handle:
            pickle.dump(my_dict, handle, protocol=pickle.HIGHEST_PROTOCOL)
        return my_dict

In [ ]:
basic_english.sort()
my_dict = make_new_dictionary()

Running...
1332


C:\Users\darin\Anaconda2\lib\site-packages\ipykernel_launcher.py:15: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.
  from ipykernel import kernelapp as app


starting...
able: 1 of 1332
    abilities -> [u'abilities', 'able', 'NNS']
    ablity -> [u'ablity', 'able', 'NN']
    ability -> [u'ability', 'able', 'NN']
    able -> [u'able', 'able', 'JJ']
    ability -> [u'ability', 'able', 'NN']
    able -> [u'able', 'able', 'JJ']
about: 2 of 1332
    abouta -> [u'abouta', 'about', 'NN']
    about -> [u'about', 'about', 'IN']
    abouts -> [u'abouts', 'about', 'NNS']
    about -> [u'about', 'about', 'IN']
    about -> [u'about', 'about', 'IN']
above: 3 of 1332
    above -> [u'above', 'above', 'IN']
    above -> [u'above', 'above', 'IN']
    above -> [u'above', 'above', 'IN']
    abover -> [u'abover', 'above', 'NN']
absence: 4 of 1332
    absents -> [u'absents', 'absence', 'NNS']
    absented -> [u'absented', 'absence', 'VBN']
    absentee -> [u'absentee', 'absence', 'NN']
    absenting -> [u'absenting', 'absence', 'VBG']
    absenteeism -> [u'absenteeism', 'absence', 'NN']
    absentees -> [u'absentees', 'absence', 'NNS']
    absence -> [u'absenc

after: 16 of 1332
    aftter -> [u'aftter', 'after', 'NN']
    after -> [u'after', 'after', 'IN']
    after -> [u'after', 'after', 'IN']
    aftre -> [u'aftre', 'after', 'NN']
    aftr -> [u'aftr', 'after', 'NN']
    afte -> [u'afte', 'after', 'NN']
    after -> [u'after', 'after', 'IN']
    aftering -> [u'aftering', 'after', 'VBG']
    after -> [u'after', 'after', 'IN']
again: 17 of 1332
    again -> [u'again', 'again', 'RB']
    again -> [u'again', 'again', 'RB']
    again -> [u'again', 'again', 'RB']
    agains -> [u'agains', 'again', 'NNS']
against: 18 of 1332
    againsts -> [u'againsts', 'against', 'NNS']
    against -> [u'against', 'against', 'IN']
    against -> [u'against', 'against', 'IN']
    against -> [u'against', 'against', 'IN']
age: 19 of 1332
    ageism -> [u'ageism', 'age', 'NN']
    ageist -> [u'ageist', 'age', 'NN']
    aging -> [u'aging', 'age', 'VBG']
    ages -> [u'ages', 'age', 'NNS']
    aged -> [u'aged', 'age', 'VBN']
    age -> [u'age', 'age', 'NN']
    age -

animal: 41 of 1332
    animals -> [u'animals', 'animal', 'NNS']
    anima -> [u'anima', 'animal', 'NN']
    animalistic -> [u'animalistic', 'animal', 'JJ']
    animal -> [u'animal', 'animal', 'NN']
    animality -> [u'animality', 'animal', 'NN']
    animal -> [u'animal', 'animal', 'NN']
    animals -> [u'animals', 'animal', 'NNS']
    animeals -> [u'animeals', 'animal', 'NNS']
    animals -> [u'animals', 'animal', 'NNS']
    animal -> [u'animal', 'animal', 'NN']
ankle: 42 of 1332
    ankles -> [u'ankles', 'ankle', 'NNS']
    ankle -> [u'ankle', 'ankle', 'NN']
    ankle -> [u'ankle', 'ankle', 'NN']
    ankle -> [u'ankle', 'ankle', 'NN']
    ankles -> [u'ankles', 'ankle', 'NNS']
    ankled -> [u'ankled', 'ankle', 'VBN']
answer: 43 of 1332
    answ -> [u'answ', 'answer', 'NN']
    answ -> [u'answ', 'answer', 'NN']
    answered -> [u'answered', 'answer', 'VBN']
    answers -> [u'answers', 'answer', 'NNS']
    answer -> [u'answer', 'answer', 'NN']
    answers -> [u'answers', 'answer', 'NNS'

as: 58 of 1332
    as -> [u'as', 'as', 'IN']
    as -> [u'as', 'as', 'IN']
    asan -> [u'asan', 'as', 'NN']
    asa -> [u'asa', 'as', 'NN']
ash: 59 of 1332
    ash -> [u'ash', 'ash', 'NN']
    ashing -> [u'ashing', 'ash', 'VBG']
    ashed -> [u'ashed', 'ash', 'VBN']
    ashen -> [u'ashen', 'ash', 'NN']
    ashes -> [u'ashes', 'ash', 'NNS']
    ash -> [u'ash', 'ash', 'NN']
ask: 60 of 1332
    askers -> [u'askers', 'ask', 'NNS']
    asks -> [u'asks', 'ask', 'NNS']
    aske -> [u'aske', 'ask', 'NN']
    ask -> [u'ask', 'ask', 'NN']
    askes -> [u'askes', 'ask', 'NNS']
    asker -> [u'asker', 'ask', 'NN']
    asked -> [u'asked', 'ask', 'VBN']
    asked -> [u'asked', 'ask', 'VBN']
    ask -> [u'ask', 'ask', 'NN']
    asking -> [u'asking', 'ask', 'VBG']
    ask -> [u'ask', 'ask', 'NN']
    asking -> [u'asking', 'ask', 'VBG']
    asked -> [u'asked', 'ask', 'VBN']
    asking -> [u'asking', 'ask', 'VBG']
    asks -> [u'asks', 'ask', 'NNS']
    asks -> [u'asks', 'ask', 'NNS']
    askance -> [u

bag: 74 of 1332
    bagging -> [u'bagging', 'bag', 'NN']
    bag -> [u'bag', 'bag', 'NN']
    baggage -> [u'baggage', 'bag', 'NN']
    bags -> [u'bags', 'bag', 'NNS']
    bagged -> [u'bagged', 'bag', 'VBN']
    bagger -> [u'bagger', 'bag', 'NN']
    bag -> [u'bag', 'bag', 'NN']
    bagging -> [u'bagging', 'bag', 'NN']
    bags -> [u'bags', 'bag', 'NNS']
    baggages -> [u'baggages', 'bag', 'NNS']
    bags -> [u'bags', 'bag', 'NNS']
    baggage -> [u'baggage', 'bag', 'NN']
    bagged -> [u'bagged', 'bag', 'VBN']
    bagful -> [u'bagful', 'bag', 'NN']
    bag -> [u'bag', 'bag', 'NN']
    bagfuls -> [u'bagfuls', 'bag', 'NN']
balance: 75 of 1332
    balancers -> [u'balancers', 'balance', 'NNS']
    balance -> [u'balance', 'balance', 'NN']
    balancer -> [u'balancer', 'balance', 'NN']
    balances -> [u'balances', 'balance', 'NNS']
    balanced -> [u'balanced', 'balance', 'VBN']
    balancing -> [u'balancing', 'balance', 'VBG']
    balances -> [u'balances', 'balance', 'NNS']
    balanced -

bell: 96 of 1332
    bell -> [u'bell', 'bell', 'NN']
    bellied -> [u'bellied', 'bell', 'VBN']
    belling -> [u'belling', 'bell', 'NN']
    belen -> [u'belen', 'bell', 'NN']
    belty -> [u'belty', 'bell', 'NN']
    belly -> [u'belly', 'bell', 'RB']
    bells -> [u'bells', 'bell', 'NNS']
    bell -> [u'bell', 'bell', 'NN']
    bell -> [u'bell', 'bell', 'NN']
    belled -> [u'belled', 'bell', 'VBN']
    bells -> [u'bells', 'bell', 'NNS']
    bells -> [u'bells', 'bell', 'NNS']
belt: 97 of 1332
    belt -> [u'belt', 'belt', 'NN']
    belter -> [u'belter', 'belt', 'NN']
    belted -> [u'belted', 'belt', 'VBN']
    belts -> [u'belts', 'belt', 'NNS']
    belt -> [u'belt', 'belt', 'NN']
    belting -> [u'belting', 'belt', 'NN']
    belt -> [u'belt', 'belt', 'NN']
    belts -> [u'belts', 'belt', 'NNS']
    beltings -> [u'beltings', 'belt', 'NNS']
beneath: 98 of 1332
    beneath -> [u'beneath', 'beneath', 'NN']
    beneath -> [u'beneath', 'beneath', 'NN']
    beneath -> [u'beneath', 'beneath'